.github/workflows/native-full_marian-ubuntu.yml

```diff
@@ -87,34 +61,89 @@ jobs:
         CC=/usr/bin/gcc-${{ matrix.gcc }} CXX=/usr/bin/g++-${{ matrix.gcc }} CUDAHOSTCXX=/usr/bin/g++-${{ matrix.gcc }} \
         cmake .. \
           -DBoost_ARCHITECTURE=-x64 \
-          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_BUILD_TYPE=Slim \
```

Comment:
Any specific reason to change it to `Slim`?

Reply:
Binaries (artifacts uploaded) are 2GB with debug symbols. It's `Slim` that provides no `-g` and hence tinier artifacts. GitHub CI doesn't require `-g` I hope. :-)

Reply:
Marian [enables NDEBUG](https://github.com/browsermt/marian-dev/blob/master/CMakeLists.txt#L266) for `Slim` builds which will remove [asserts()](https://en.cppreference.com/w/cpp/error/assert) in your binaries.

I understand your problem though. May be keep a release build and a separate workflow for slim builds just for the regression tests?

I am not sure what is the right call here.

Reply:
`Release` is set by Marcin in an attempt to minimize support e-mails.  IMHO Release should not have assertions or `-g` either.  But I agree it's good to have perhaps a separate job with less expensive testing that verifies assertions are still met.

Reply:
> I agree it's good to have perhaps a separate job with less expensive testing that verifies assertions are still met.

So I clubbed the regression test into the same workflow avoiding the problem of big binaries. In addition, the native minimal marian config has been collapsed (so we'll only have two workflow files sharing the specs for regression-tests and install, varying cmake build in the matrix strategy). Sorry for the additional review load, but this should move the state of tests to an overall better state I hope. 

I request @abhi-agg to revisit https://github.com/browsermt/bergamot-translator-tests/pull/7 (I've pushed a few changes behind the previous state, and we're now ready to merge this). I'm placing the the README update behind this.

With this, I'm also allowing a tag-based regression-testing for the minimal/custom marian (testing only bergamot-translator-app, switching the build by ifdefs through CMake options).

Reply:
> I request @abhi-agg to revisit [browsermt/bergamot-translator-tests#7](https://github.com/browsermt/bergamot-translator-tests/pull/7) (I've pushed a few changes behind the previous state, and we're now ready to merge this).

Approved 👍 

> I'm placing the the README update behind this.

Awesome 👍